tests/unit/single_table/test_base.py

```diff
@@ -61,3 +69,328 @@ def test_get_metadata(self, mock_data_processor):
 
         # Assert
         assert result == metadata
+
+    def test_validate_type(self):
+        """When data is not of type ``pd.DataFrame``, an error should be raised."""
```

Comment:
I think eventually we might run the lint on tests as well and they require docstrings to be in an imperative mood.